htdocs/install/mysql/migration/17.0.0-18.0.0.sql

```diff
@@ -476,3 +476,67 @@ ALTER TABLE llx_actioncomm ADD INDEX idx_actioncomm_ref (ref, entity);
 -- Bump llx_reception.ref_supplier to allow up to 255 characters to match llx_commande_fournisseur.ref_supplier.
 -- See: https://github.com/Dolibarr/dolibarr/pull/25034
 ALTER TABLE llx_reception MODIFY COLUMN ref_supplier varchar(255);
+
+-- Use foreign key contraints for extrafields
+ALTER TABLE llx_actioncomm_extrafields DROP INDEX idx_actioncomm_extrafields; ALTER TABLE llx_actioncomm_extrafields ADD FOREIGN KEY idx_actioncomm_extrafields (fk_object) REFERENCES llx_actioncomm (id);
```

Comment:
Index and foreign keys are 2 differents thinks. Index is adding data managed for performance (idx_...)
Forrign keys is a rule/constraint to check (called fk_...)
Note tha adding a foreign keys needs an index key on the parent field

Reply:
Thanks for pointing this out.  Will the following be good, or should I avoid the foreign key?

```
+ALTER TABLE llx_actioncomm_extrafields DROP INDEX idx_actioncomm_extrafields; ALTER TABLE llx_actioncomm_extrafields ADD UNIQUE INDEX uk_actioncomm_extrafields (fk_object); ALTER TABLE llx_actioncomm_extrafields ADD FOREIGN KEY fk_actioncomm_extrafields (fk_object) REFERENCES llx_actioncomm (id);
```

Reply:
There are two parts to this merge request:

1) Use of UNIQUE INDEX to ensure that there is no more than one extrafields row for every master table row.  Without this it is more work to prevent duplicate extrafields rows, where the authoritative value for a field could be uncertain.  This is important.  (First of two commits in this MR)

2) Use of FOREIGN KEY to ensure that every extrafields row corresponds to a master table row.  This is not critical and could be omitted from this merge request.  (Second of two commits in this MR)